src/bin/nydusd/fusedev.rs

```diff
@@ -175,14 +176,20 @@ impl FusedevDaemon {
         let thread = thread::Builder::new()
             .name("fuse_server".to_string())
             .spawn(move || {
-                let _ = s.svc_loop(&inflight_op);
-                // quit the daemon if any fuse server thread exits
-                exit_daemon();
+                if let Err(err) = s.svc_loop(&inflight_op) {
+                    warn!("fuse server exit with err: {:?}, exit daemon", err);
+                    if let Err(e) = thread::Builder::new()
+                        .name("exit_thread".to_string())
+                        .spawn(exit_daemon)
```

Comment:
I still don't catch the idea why the `exit_daemon` has to be invoked asynchronously.
Looks that nydusd's main thread does nothing eventually but wait for state machine and fuse service loop termination. `exit_daemon` only pushes `Stop` event. Can we just wait for statemachine thread here.  And we stop state machine by a carefully predefined event  `StopStateMachine`, which must not involve blocking operations. Then we don't need to spawn more threads which looks a little weird.

Reply:
> I still don't catch the idea why the `exit_daemon` has to be invoked asynchronously. Looks that nydusd's main thread does nothing eventually but wait for state machine and fuse service loop termination. `exit_daemon` only pushes `Stop` event. Can we just wait for statemachine thread here. And we stop state machine by a carefully predefined event `StopStateMachine`, which must not involve blocking operations. Then we don't need to spawn more threads which looks a little weird.

The worker thread is responsibility to yell 'F**k by unexpected error', however without responsibility to guarantee   process exiting. If state machine or processs crews up, the worker threads still exits. The worker takes no care about whether the state machine or process exits or not.
But the way to throw out signal is absolutely weird by spawning new thread. Maybe change other async way in other PR, such as channel in state machine thread, SYSTERM/SYSINT.